### DISCUSSION TITLE
What's the common features to cluster?

### DISCUSSION THREAD

**Discussion by pahome** [blue-yonder/tsfresh#860] [Q&A]
I use ubuntu14.4 desktop and try to cluster time-series dataset.

When I use tsfresh.extract_features, I get multiple features to cluster like  number_peaks__n_3, __maximum...etc.

There're too many features to know what is effecient for me because I do unsupervised learning so I don't have y for tslearn.select_features(X,y)

Are there any common features to cluster?

**Comment by MaxBenChrist** [comment]
> Are there any common features to cluster?

Not that I am aware of. 
But, maybe you could start with some basic features (min, max, mean etc.) to get quick results.

Maybe use 
https://tsfresh.readthedocs.io/en/latest/api/tsfresh.feature_extraction.html#tsfresh.feature_extraction.settings.MinimalFCParameters 
to get some basic features for quick clustering quick results and then later add more and more complex features?

**Comment by nils-braun** [comment]
Hm, it is very hard to answer that in general - it depends on what you definition of a cluster should be?

Let me give you an example: lets look into stock market prices. The absolute value is not a good variable to cluster on - we are normally more interested in clustering stocks with a "similar behaviour" together. So maybe it is FFT parameters or alike in this case.
However look at another example: industry 4.0 in a plant: temperature time series of a manufactured piece. If the temperature is too high or too cold, the piece might be bad. So here the maximum value (= maximum temperature) can be a very good clustering feature.

You see, it really depends on your data and a lot on your domain knowledge. Clustering based on non-sense features for your use case might be even bad....

**Comment by kempa-liehr** [comment]
@pahome: We have recently published a paper on unsupervised feature selection, which might be useful for your application: https://doi.org/10.1109/JSEN.2021.3084970
